### PR TITLE
Implementing Shimmer Loading Effect in Android Studio

### LINKED ISSUE #1069: Implementing Shimmer Loading Effect in Android Studio
### Brief Summary: 
[Shimmer](https://facebook.github.io/shimmer-android/) is an Android library that implements a shimmer loading effect (Skeleton Screen Loading) when fetching data from a server. Facebook developed it to show loading animation while loading data from API or the Local Database. It implements the same concept as you would do with loading or a progress dialog. Many big corporates such as YouTube, Linkedin, Google Drive and Facebook have implemented the shimmer loading effect, just to name a few.
shimmer-loading-effect

### Key Takeaways: 

This guide will discuss how to implement the loading effect using android studio using live data. We will use HTTP request libraries such as [volley](https://www.section.io/engineering-education/making-api-requests-using-volley-android/) or [retrofit](https://github.com/section-io/engineering-education/pull/989) to load data from the internet and display it the app page while implementing the shimmer loading effect. The guide will not focus on the volley or retrofit section have more information about them. The guide will focus on shimmer effects.

### References: 
https://facebook.github.io/shimmer-android/

### PR DESCRIPTION
**Pull request by kimkimani**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
Fixes #1069

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Hey @kimkimani  I just finished the preliminary review of the article. It is well structured and easy to understand. Could you make the following minor changes or clarifications.

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 1]
```diff
@@ -533,7 +546,7 @@ public class MainActivity extends AppCompatActivity {
 }
 ```
 
-As you can see above, the code implements loading data from the internet using the Volley library into the RecyclerView. Here we load the data in the JSON URL we included above. We load the data and display it into the main activity. In this case, we load the RecyclerView content into the main screen of our application.
+As you can see above, the code implements loading data from the Internet using the Volley library into the RecyclerView. Here we load the data in the JSON URL we included above. We load the data and display it into the main activity. In this case, we load the RecyclerView content into the main screen of our application.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:549]: Could you some synonyms for the word load -- loading --loaded. It seems to be overused

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 2]
```diff
@@ -333,9 +345,10 @@ Here the shimmer layout design.
 
 ### Putting in place the right classes and functions to load data and shimmer effect into the RecyclerView
 
-#### The view model
+#### The model
 
-The primary purpose of view model classes is to host the objects that hold the data to be displayed in a view. Go ahead and create a `Students.java` class and include the following code block in it.
+The primary purpose of the model class is to host the objects that hold the data to be displayed in a view.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:350]: Could you clarify more on the purpose of the model class

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
 
-To implement a shimmering effect in android studio, you need a library called [shimmer](http://facebook.github.io/shimmer-android/). This dependency helps implement a shimmer layouts when fetching data from a server. It implements the same concept as you would do with Spinner or ProgressBar. Major corporates such as YouTube, LinkedIn, Google Drive, and Facebook have implemented the shimmer loading effect in their applications. [Facebook]((http://facebook.github.io/shimmer-android/)) developed it to show loading animation while requesting data from APIs or servers.
+To implement a shimmering effect in android studio, you need a library called [Shimmer](http://facebook.github.io/shimmer-android/). This dependency helps implement a shimmer layout when fetching data from a server. It implements the same concept as a Spinner or ProgressBar. Major corporates such as YouTube, LinkedIn, Google Drive, and Facebook have implemented the shimmer loading effect in their applications. [Facebook]((http://facebook.github.io/shimmer-android/)) developed it to show loading animation while requesting data from APIs or servers.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:9]: Could you please place them in a list format

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 4]
```diff
@@ -81,18 +83,22 @@ Since the application is making remote calls, add internet permissions into your
 We are ready to set the Shimmering effect in the RecyclerView layout. This application involves two phases.
 
 - Loading data into the view
-- Shimmering the data loading process.
+- Show a shimmering animation when fetching data.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:86]: Which one happens first? Could you please rearrange them

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 5]
```diff
@@ -185,11 +191,11 @@ Here is the code to implement that.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:191]: No need for the extra <androidx.constraintlayout.widget.ConstraintLayout/> in the above layout

File: articles/android-studio-shimmer-loading-effect/index.md [hunk 6]
```diff
@@ -283,9 +289,9 @@ Here is the shimmer layout design.
```
Comment (WanjaMIKE) [inline articles/android-studio-shimmer-loading-effect/index.md:289]: android:orientation="vertical" does not have an effect on the parent RelativeLayout. Its mainly for LinearLayout. Could you please remove it.

**Review by WanjaMIKE — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**